src/components/dialog-parameters/single-line-diagram-parameter.js

```diff
@@ -0,0 +1,124 @@
+/**
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+import React, { useEffect, useState } from 'react';
+
+import { FormattedMessage } from 'react-intl';
+
+import { Grid, MenuItem, Box, Select, Typography } from '@mui/material';
+
+import { getAvailableComponentLibraries } from '../../utils/rest-api';
+import { SubstationLayout } from '../diagrams/singleLineDiagram/single-line-diagram';
+import {
+    PARAM_CENTER_LABEL,
+    PARAM_DIAGONAL_LABEL,
+    PARAM_SUBSTATION_LAYOUT,
+    PARAM_COMPONENT_LIBRARY,
+} from '../../utils/config-params';
+import { LabelledSwitch, useParameterState } from './parameters';
+import { LineSeparator } from './line-separator';
+import { useStyles } from './parameters';
+
+export const SingleLineDiagramParameters = ({ user }) => {
+    const classes = useStyles();
+
+    const [diagonalLabelLocal, handleChangeDiagonalLabel] =
+        useParameterState(PARAM_DIAGONAL_LABEL);
+    const [centerLabelLocal, handleChangeCenterLabel] =
+        useParameterState(PARAM_CENTER_LABEL);
+    const [substationLayoutLocal, handleChangeSubstationLayout] =
+        useParameterState(PARAM_SUBSTATION_LAYOUT);
+    const [componentLibraryLocal, handleChangeComponentLibrary] =
+        useParameterState(PARAM_COMPONENT_LIBRARY);
+    const [componentLibraries, setComponentLibraries] = useState([]);
+    useEffect(() => {
+        if (user !== null) {
+            getAvailableComponentLibraries().then((libraries) => {
+                setComponentLibraries(libraries);
+            });
+        }
+    }, [user]);
+
+    return (
+        <Grid container spacing={1} className={classes.grid}>
+            {LabelledSwitch(diagonalLabelLocal, 'diagonalLabel', () => {
```

Comment:
now can be written 
`<LabelledSwitch prop={diagonalLabelLocal} label='diagonalLabel' callback=() => ...`